Instructions/Exercises/01-prompt-engineering.md

```diff
@@ -0,0 +1,292 @@
+
+This exercise will take approximately **30** minutes.
+
+## Provision an Azure OpenAI resource
+
+If you don't already have one, provision an Azure OpenAI resource in your Azure subscription.
+
+1. Sign into the **Azure portal** at `https://portal.azure.com`.
+
+1. Create an **Azure OpenAI** resource with the following settings:
+    - **Subscription**: *Select an Azure subscription that has been approved for access to the Azure OpenAI service*
+    - **Resource group**: *Choose or create a resource group*
+    - **Region**: *Make a **random** choice from any of the following regions*\*
+        - Canada East
+        - East US
+        - East US 2
+        - France Central
+        - Japan East
+        - North Central US
+        - Sweden Central
+        - Switzerland North
+        - UK South
+    - **Name**: *A unique name of your choice*
+    - **Pricing tier**: Standard S0
+
+    > \* Azure OpenAI resources are constrained by regional quotas. The listed regions include default quota for the model type(s) used in this exercise. Randomly choosing a region reduces the risk of a single region reaching its quota limit in scenarios where you are sharing a subscription with other users. In the event of a quota limit being reached later in the exercise, there's a possibility you may need to create another resource in a different region.
+
+3. Wait for deployment to complete. Then go to the deployed Azure OpenAI resource in the Azure portal.
+
+## Deploy a model
+
+Next, you will deploy an Azure OpenAI model resource from the CLI. Refer to this example and replace the following variables with your own vales from above:
+
+```dotnetcli
+az cognitiveservices account deployment create \
+   -g *Your resource group* \
+   -n *Name of your OpenAI service* \
+   --deployment-name gpt-35-turbo \
+   --model-name gpt-35-turbo \
+   --model-version 0125  \
+   --model-format OpenAI \
+   --sku-name "Standard" \
+   --sku-capacity 5
+```
+
+    > \* Sku-capacity is measured in thousands of tokens per minute. A rate limit of 5,000 tokens per minute is more than adequate to complete this exercise while leaving capacity for other people using the same subscription.
+
+## Configure your application
+
+Applications for both C# and Python have been provided, and both apps feature the same functionality. First, you'll complete some key parts of the application to enable using your Azure OpenAI resource with asynchronous API calls.
+
+1. In Visual Studio Code, in the **Explorer** pane, browse to the **Labfiles/xx-develop-generative-ai** folder and expand the **CSharp** or **Python** folder depending on your language preference. Each folder contains the language-specific files for an app into which you're you're going to integrate Azure OpenAI functionality.
```

Comment:
the navigation says to go to `xx-develop-generative-ai` folder, but it appears to be still named prompt engineering? can you update the folder/program names to match the new module and make sure this instruction says the right folder name?

Reply:
fixed